Modix.Services/Core/DesignatedChannelService.cs

```diff
@@ -0,0 +1,179 @@
+        Task<IEnumerable<IMessageChannel>> GetDesignatedChannels(IGuild guild, ChannelDesignation designation);
+
+        /// <summary>
+        /// Retrieves a collection of <see cref="ulong"/>s corresponding to the channels assigned to a given designation, for a given guild id.
+        /// </summary>
+        /// <param name="guild">The ID of the guild to retrieve designations for</param>
+        /// <param name="designation">The designation to retrieve assigned channels for</param>
+        /// <returns>A <see cref="Task"/> that will complete when all <see cref="IMessageChannel"/>s have been retrieved.</returns>
+        Task<IEnumerable<ulong>> GetDesignatedChannelIds(ulong guildId, ChannelDesignation designation);
+
+        /// <summary>
+        /// Retriveves a collection of <see cref="DesignatedChannelMappingBrief"/>s associated with the given guild.
+        /// </summary>
+        /// <param name="guildId">The ID of the guild to retrieve designations for</param>
+        /// <returns>A <see cref="Task"/> that will complete when all <see cref="DesignatedChannelMappingBrief"/>s have been retrieved.</returns>
+        Task<IEnumerable<DesignatedChannelMappingBrief>> GetDesignatedChannels(ulong guildId);
+
+        /// <summary>
+        /// Sends the given message (and embed) to the <see cref="IMessageChannel"/>s assigned to the given designation, for a given guild.
+        /// </summary>
+        /// <param name="guild">The <see cref="IGuild"/> to send the message to</param>
+        /// <param name="designation">The <see cref="ChannelDesignation"/> of the channels to send the messages to</param>
+        /// <param name="content">The text content of the message</param>
+        /// <param name="embed">An optional <see cref="Embed"/> to attach to the message</param>
+        /// <returns>A <see cref="Task"/> that, when completed, results in a collection of the messages that were sent.</returns>
+        Task<IEnumerable<IMessage>> SendToDesignatedChannelsAsync(IGuild guild, ChannelDesignation designation, string content, Embed embed = null);
+    }
+
+    public class DesignatedChannelService : IDesignatedChannelService
+    {
+        internal protected IDesignatedChannelMappingRepository DesignatedChannelMappingRepository { get; }
+        internal protected IAuthorizationService AuthorizationService { get; }
+
+        public DesignatedChannelService(IDesignatedChannelMappingRepository designatedChannelMappingRepository, IAuthorizationService authorizationService)
+        {
+            DesignatedChannelMappingRepository = designatedChannelMappingRepository;
+            AuthorizationService = authorizationService;
+        }
+
+        /// <inheritdoc />
+        public async Task AddDesignatedChannelAsync(IGuild guild, IMessageChannel logChannel, ChannelDesignation designation)
+        {
+            AuthorizationService.RequireClaims(AuthorizationClaim.ChannelDesignationCreate);
```

Comment:
Ah, I already refactored `RequireClaims()` to call `RequireAuthenticatedUser()` itself in a previous PR. I can revert that if it doesn't make sense.

Reply:
.............................................................................................. I must have missed that.

I dunno, I like the idea of keeping it separate, since the code doesn't literally require a user to populate the claims collection. If it's already been changed, I can give no reason to change it back. If we ever DID need to separate them, we can refactor at that time.